libs/novui/src/json-schema-components/widgets/InputEditorWidget.tsx

```diff
@@ -0,0 +1,134 @@
+
+export const InputEditorWidget = (props: WidgetProps) => {
+  const { value, label, formContext, onChange, required, readonly, rawErrors, options, schema } = props;
+  const [variantProps, inputProps] = input.splitVariantProps({});
+  const [cssProps] = splitCssProps(inputProps);
+  const classNames = input(variantProps);
+
+  const { variables = [] } = formContext;
+  const reactRenderer = useRef<ReactRenderer<SuggestionListRef>>(null);
+
+  const variablesList = useMemo<VariableItem[]>(() => {
+    return variables?.map((variable: string) => {
+      return { label: variable, id: variable };
+    });
+  }, [variables]);
+
+  const baseExtensions: Extensions = [Document, Paragraph, Text];
+
+  if (variables.length) {
+    baseExtensions.push(
+      Mention.configure({
+        HTMLAttributes: {
+          class: 'suggestion',
+        },
+        renderHTML: ({ options, node }) => {
+          return [
+            'span',
+            options.HTMLAttributes,
+            `${options.suggestion.char}${node.attrs.label ?? node.attrs.id}${AUTOCOMPLETE_CLOSE_TAG}`,
+          ];
+        },
+        renderText: ({ options, node }) => {
+          return `${options.suggestion.char}${node.attrs.label ?? node.attrs.id}${AUTOCOMPLETE_CLOSE_TAG}`;
+        },
+        suggestion: {
+          items: ({ query }) => {
+            return variablesList?.filter((item) => item.label.toLowerCase().includes(query.toLowerCase()));
+          },
+          char: AUTOCOMPLETE_OPEN_TAG,
+          render() {
+            return {
+              onStart: (props) => {
+                reactRenderer.current = new ReactRenderer(VariableSuggestionList, {
+                  props,
+                  editor: props.editor,
+                });
+              },
+              onUpdate(props) {
+                reactRenderer.current?.updateProps(props);
+              },
+              onKeyDown(props) {
+                if (!reactRenderer.current?.ref) {
+                  return false;
+                }
+
+                return reactRenderer.current?.ref.onKeyDown(props);
+              },
+              onExit() {
+                reactRenderer.current?.destroy();
+              },
+            };
+          },
+        },
+      })
+    );
+  }
+
+  const editor = useEditor({
+    extensions: baseExtensions,
+    content: '',
+    editable: !readonly,
+    onFocus: () => {
+      reactRenderer.current?.ref?.focus();
+    },
+    onUpdate: ({ editor }) => {
+      const content = editor.isEmpty ? undefined : editor.getText();
+      onChange(content);
+    },
+  });
+
+  useEffect(() => {
+    if (editor) {
+      const newValue = value
+        ?.toString()
+        .replace(
+          AUTOCOMPLETE_REGEX,
+          '<span data-id="$1" contenteditable="false" class="suggestion" data-type="mention">$1</span>'
+        );
+
+      editor.commands.setContent(newValue);
+    }
+  }, []);
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** Should we include `editor` here as a dependency? If it breaks things, then do we need an eslint-disable here for dependencies here? ‏